### ISSUE TITLE
Time-series cross validation

### ISSUE THREAD

**Issue by gcolitti** [paul-buerkner/brms#608]
Currently, `loo` and `kfold` facilitate model comparison for models without a time dependency. To my knowledge, brms does not support time-series cross-validation in the same way. Are methods such as approximate leave-future-out cross-validation (http://mc-stan.org/loo/articles/loo2-lfo.html), fixed window rolling forecast, or increasing window rolling forecast features that might be implemented in the future? For instance, I'm thinking `kfold` might be augmented to include support for rolling forecasts through (maybe) the `folds` option. Perhaps something along the lines of: `folds = "rolling"`. Further arguments that determine the initial window size, horizon, and whether the window is fixed would add further value. Hoping this post isn't out of place. Thanks!

**Comment by paul-buerkner** [comment]
We actually have a full paper on approximate leave-future-out cross-validation available on arXiv now (https://arxiv.org/abs/1902.06281) and I am definitely planning on implementing this, and perhaps other methods in brms as soon as I find the time. @avehtari any thoughs on this?

**Comment by jgabry** [comment]
Definitely in favor of implementing the LFO-CV stuff at some point. I think we should try to build as much of the time series stuff into the loo package as possible so that all the R packages can share to the extent possible.

**Comment by paul-buerkner** [comment]
I agree. @topipa and I are doing the same with the moment matching loo method, that is abstracting away all the specifics so that the main algorithm can go into **loo**.
